src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js

```diff
@@ -160,14 +160,47 @@ class ActivityFeed extends React.Component<Props, State> {
         versionInfoHandler(data);
     };
 
+    /**
+     * Determine whether or not a sort should occur, based on new comments, tasks, versions. Also
+     * clears the feed state of old data if the file has been changed.
+     *
+     * @param {Comments} - [comments] - Object containing comments for the file.
+     * @param {Tasks} - [tasks] - Object containing tasks for the file.
+     * @param {FileVersions} - [versions] Object containing versions of the file.
+     * @param {BoxItem} - [file] The box file that comments, tasks, and versions belong to.
+     * @return {boolean} True if the feed should be sorted with new items.
+     */
+    shouldSortFeedItems(comments?: Comments, tasks?: Tasks, versions?: FileVersions, file?: BoxItem): boolean {
+        if (!comments || !tasks || !versions) {
+            return false;
+        }
+
+        // Reset feed items if the file has changed
+        const { file: oldFile } = this.props;
+        if (file && file.id !== oldFile.id) {
+            this.setState({ feedItems: [] });
+        }
+
+        return true;
+    }
+
     componentDidMount(): void {
-        const { comments, tasks, versions } = this.props;
-        this.sortFeedItems(comments, tasks, versions);
+        const { comments, tasks, versions, file } = this.props;
+        const shouldSort = this.shouldSortFeedItems(comments, tasks, versions, file);
+        if (shouldSort) {
+            // $FlowFixMe
+            this.sortFeedItems(comments, tasks, versions);
+        }
     }
 
     componentWillReceiveProps(nextProps: any): void {
-        const { comments, tasks, versions } = nextProps;
-        this.sortFeedItems(comments, tasks, versions);
+        const { comments, tasks, versions, file } = nextProps;
+        const { feedItems } = this.state;
+        const shouldSort = this.shouldSortFeedItems(comments, tasks, versions, file);
+        if (shouldSort && !feedItems.length) {
```

Comment:
Am not 100% sure about this. setState is technically async and takes a callback. Maybe add the ```sortFeedItems``` into the setState callback?